File: app/src/main/java/c/m/storyapp/list_story/data/remote/dto/ListStoryResponseDTO.kt
package c.m.storyapp.list_story.data.remote.dto

import com.google.gson.annotations.SerializedName

data class ListStoryResponseDTO(
    @SerializedName("error")
    val error: Boolean,
    @SerializedName("message")
    val message: String,
    @SerializedName("listStory")
    val listStory: List<ListStoryDTO>
)

File: lib/list_story/data/remote/dto/list_story_response_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:story_app/list_story/data/remote/dto/story_response_dto.dart';

part 'list_story_response_dto.freezed.dart';
part 'list_story_response_dto.g.dart';

@freezed
class ListStoryResponseDTO with _$ListStoryResponseDTO {
  const factory ListStoryResponseDTO({
    @JsonKey(name: 'error') required bool error,
    @JsonKey(name: 'message') required String message,
    @JsonKey(name: 'listStory') List<StoryResponseDTO>? listStoryResponseDTO,
  }) = _ListStoryResponseDTO;

  factory ListStoryResponseDTO.fromJson(Map<String, dynamic> json) =>
      _$ListStoryResponseDTOFromJson(json);
}
